dipy/viz/actor.py

```diff
@@ -64,46 +68,40 @@ def slicer(data, affine=None, value_range=None, opacity=1.,
     else:
         nb_components = 1
 
-    if value_range is None:
-        vol = np.interp(data, xp=[data.min(), data.max()], fp=[0, 255])
-    else:
-        vol = np.interp(data, xp=[value_range[0], value_range[1]], fp=[0, 255])
-    vol = vol.astype('uint8')
-
+    # Allocate VTK Image Data
+    data = np.copy(data)
     im = vtk.vtkImageData()
-    if major_version <= 5:
-        im.SetScalarTypeToUnsignedChar()
-    I, J, K = vol.shape[:3]
+    if nb_components == 1:
+        if value_range is None:
+            value_range = (np.min(data), np.max(data))
+        else:
+            # Clamping data to specified value range
+            data[data < value_range[0]] = value_range[0]
```

Comment:
I think here we have a different understanding of what this function should be doing. I wanted this dataset to interpolate  from value_range to (0, 255). I think you want to clamp the data. That is a different goal.